### PR TITLE
Fixes

### PR DESCRIPTION
**Pull request by limitedeternity**
Resolves [#147](https://github.com/dutchcoders/transfer.sh/issues/147) & [#148](https://github.com/dutchcoders/transfer.sh/issues/148)

Adds embedding functionality using `content-disposition: inline; filename=`

### CODE REVIEW

**Comment by limitedeternity** [comment]
1. Is `/get/...` intended to be an alias for `/download/...`?
If so, I'll add this line back.

2. Yep, I agree that `inline` is more suitable here than `embed`. 

3. Creating `Link aliases` section seems good idea for me.

4. `HEAD` requests are useful for "pinging" resources without actually getting them. Without `HEAD` support we need to send `GET` request which results in whole 1GB+ file (for example) downloaded. That's quite unwanted behavior if you only need to check response's status code (for instance, to check if remote file still exists).

5. Duplicating `getHandler` isn't good approach indeed. Will change that soon.

**Comment by aspacca** [comment]
4. I was probably not so clear about HEAD request support: I find it very useful and it's welcomed. I'm not sure if adding HEAD support to every urls or only the default one

**Comment by limitedeternity** [comment]
Like that. `Action` variable covers `download`, `get`, `inline`, `retrieve` etc. This allows us to change `content-disposition` in `getHandler` accordingly.

Also made "Link aliases" section in readme.

File: server/server.go [hunk 1]
```diff
@@ -324,6 +324,9 @@ func (s *Server) Run() {
 	r.HandleFunc("/({files:.*}).tar", s.tarHandler).Methods("GET")
 	r.HandleFunc("/({files:.*}).tar.gz", s.tarGzHandler).Methods("GET")
 
+	r.HandleFunc("/{token}/{filename}", s.headHandler).Methods("HEAD")
+	r.HandleFunc("/{action}/{token}/{filename}", s.headHandler).Methods("HEAD")
```
Comment (aspacca) [inline server/server.go]: please. change to `/{action:(download|get|inline)}/{token}/{filename}`
I'd rather prefer to limit the aliases

File: server/server.go [hunk 2]
```diff
@@ -352,8 +355,7 @@ func (s *Server) Run() {
 	}
 
 	r.HandleFunc("/{token}/{filename}", getHandlerFn).Methods("GET")
-	r.HandleFunc("/get/{token}/{filename}", getHandlerFn).Methods("GET")
-	r.HandleFunc("/download/{token}/{filename}", getHandlerFn).Methods("GET")
+	r.HandleFunc("/{action}/{token}/{filename}", getHandlerFn).Methods("GET")
```
Comment (aspacca) [inline server/server.go]: same as above

File: server/handlers.go [hunk 3]
```diff
@@ -817,8 +845,13 @@ func (s *Server) getHandler(w http.ResponseWriter, r *http.Request) {
 
 	w.Header().Set("Content-Type", contentType)
 	w.Header().Set("Content-Length", strconv.FormatUint(contentLength, 10))
-	w.Header().Set("Content-Disposition", fmt.Sprintf("attachment; filename=\"%s\"", filename))
-	w.Header().Set("Connection", "close")
+	w.Header().Set("Connection", "keep-alive")
+
+	if action == "inline" {
```
Comment (aspacca) [inline server/handlers.go]: a little redundant. why not setting a `disposition` variable having either `inline` or `attachment` as value and format it in a single `fmt.Sprintf("%s; filename=\"%s\"", disposition, filename)`?

**Comment by limitedeternity** [comment]
Should be alright now.

**Comment by limitedeternity** [comment]
Oh, well, hold on...

**Comment by limitedeternity** [comment]
Well, it's not me. Seems to be a build timeout.

**Comment by aspacca** [comment]
please, push another commit so to restart the build

**Comment by aspacca** [comment]
also it must be: `/{action:(?:download|get|inline)}/{token}/{filename}` not `/{action:(?:(download|get|inline))}/{token}/{filename}`

**Comment by limitedeternity** [comment]
Ah, `Only non-capturing groups are accepted`...

**Comment by limitedeternity** [comment]
Done!

**Pull request merged**